internal/backends/postgresql/postgresql.go

```diff
@@ -80,6 +71,16 @@ func collectionsStats(ctx context.Context, p *pgxpool.Pool, dbName string, list
 	// initialization fork https://www.postgresql.org/docs/current/storage-init.html,
 	// free space map https://www.postgresql.org/docs/current/storage-fsm.html and
 	// TOAST https://www.postgresql.org/docs/current/storage-toast.html.
+
+	return &s, placeholders, args, nil
+}
+
+func AnalyzeColl(ctx context.Context, p *pgxpool.Pool, s stats, placeholders []string, args []any) error {
+	var err error
+	q := `ANALYZE`
+	if _, err = p.Exec(ctx, q); err != nil {
+		return lazyerrors.Error(err)
+	}
 	q = fmt.Sprintf(`
```

Comment:
That code should be in collectionsStats

Reply:
But shouldn't the `ANALYZE` be isolated from the `collectionStats` method, because as long as the `ANALYZE` cmd exists as `q` in the `collectionStats` , it will be called for the same number of times as before and adding the `params.Refresh : true` check would be pointless ? i'm having trouble understanding
```
if params.Refresh{
     //Calls ANALYZE
}
stats , err := collectionStats(........) // This will call ANALYZE as well
```

Reply:
> collectionStats(........) // This will call ANALYZE as well

`collectionStats` should not call `ANALYZE`

Reply:
And that's exactly why we work on the issue titled "Call `ANALYZE` _less often_" :)